### PR TITLE
Ot159 20

### PR DESCRIPTION
**Pull request by JuanAvalo**
Added model, migrations and base controllers of Testimonials
Includes Services and Repositories

Testimonials model fields:
id: INTEGER NOT NULL AUTO_INCREMENT
name: VARCHAR NOT NULL
image: VARCHAR NULLABLE
content: VARCHAR NULLABLE
timestamps y softDeletes

### CODE REVIEW

File: controllers/testimonial.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const testimonialService = require("../services/testimonial");
+
+listAll = async (req, res) => {
+  try {
+    const testimonials = await testimonialService.listAll();
+    return res.status(200).send(testimonials);
+  } catch (error) {
+    res.status(400).send(error);
+  }
```
Comment (maxxCeballos) [inline controllers/testimonial.js:9]: No se solicita en el ticket

File: controllers/testimonial.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+findOne = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const testimonial = await testimonialService.findOne(id);
+    if (!testimonial) return res.status(404).send("Nothing found");
+
+    return res.status(200).send(testimonial);
+  } catch (error) {
+    res.status(400).send(error);
+  }
```
Comment (maxxCeballos) [inline controllers/testimonial.js:21]: No se solicita en ticket

File: migrations/20220321183657-create-testimonial.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable('Testimonials', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      name: {
+        type: Sequelize.STRING
```
Comment (maxxCeballos) [inline migrations/20220321183657-create-testimonial.js:12]: name no puede ser null

File: repositories/testimonial.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const models = require("../models");
+
+listAll = async () => {
+  const testimonials = await models.Testimonial.findAll();
+  return testimonials;
+};
+findOne = async (id) => {
+  const testimonial = await models.Testimonial.findByPk(id);
+  return testimonial;
+};
```
Comment (maxxCeballos) [inline repositories/testimonial.js:10]: No se solicita en ticket

File: services/testimonial.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const testimonialRepository = require("../repositories/testimonial");
+
+listAll = () => {
+  return testimonialRepository.listAll();
+};
+
+findOne = (id) => {
+  return testimonialRepository.findOne(id);
+};
```
Comment (maxxCeballos) [inline services/testimonial.js:9]: no se solicita en ticket

**Review by maxxCeballos — CHANGES_REQUESTED** [review]

File: migrations/20220321183657-create-testimonial.js (hunk 3 above)
Comment (JuanAvalo) [reply migrations/20220321183657-create-testimonial.js]: se agrego allowNull: false

File: controllers/testimonial.js (hunk 1 above)
Comment (JuanAvalo) [reply controllers/testimonial.js]: removido

**Review by maxxCeballos — APPROVED** [review]

**Pull request merged**